src/storage/src/storage/read_object.rs

```diff
@@ -585,6 +609,68 @@ fn check_crc32c_match(crc32c: u32, response: Option<u32>) -> Result<()> {
     Ok(())
 }
 
+fn response_range(response: &reqwest::Response) -> std::result::Result<ReadRange, ReadError> {
+    match response.status() {
+        reqwest::StatusCode::OK => {
+            let header = required_header(response, "content-length")?;
+            let limit = header
+                .parse::<u64>()
+                .map_err(|e| ReadError::BadHeaderFormat("content-length", e.into()))?;
+            Ok(ReadRange { start: 0, limit })
+        }
+        reqwest::StatusCode::PARTIAL_CONTENT => {
+            let header = required_header(response, "content-range")?;
+            let header = header.strip_prefix("bytes ").ok_or_else(|| {
+                ReadError::BadHeaderFormat("content-range", "missing bytes prefix".into())
+            })?;
+            let (range, _) = header.split_once('/').ok_or_else(|| {
+                ReadError::BadHeaderFormat("content-range", "missing / separator".into())
+            })?;
+            let (start, end) = range.split_once('-').ok_or_else(|| {
+                ReadError::BadHeaderFormat("content-range", "missing / separator".into())
+            })?;
+            let start = start
+                .parse::<u64>()
+                .map_err(|e| ReadError::BadHeaderFormat("content-range", e.into()))?;
+            let end = end
+                .parse::<u64>()
+                .map_err(|e| ReadError::BadHeaderFormat("content-range", e.into()))?;
+            let limit = end
+                .checked_sub(start)
+                .ok_or_else(|| ReadError::BadHeaderFormat("content-range", format!("range start ({start}) should be less than or equal to the range end ({end})").into()))?;
+            // HTTP ranges are inclusive, but it is more idiomatic to use a non-inclusive end point.
+            Ok(ReadRange { start, limit })
```

Comment:
This comment makes me think you are going to add 1 to `limit`

Reply:
Yes, fixed.